vortex-serde/src/file/read/stream.rs

```diff
@@ -275,9 +270,31 @@ impl<R: VortexReadAt + Unpin + 'static> VortexFileArrayStream<R> {
             StreamingState::EndOfStream => finished(),
         }
     }
+
+    /// Schedule an asynchronous read of several byte ranges.
+    ///
+    /// IO is scheduled on the provided IO dispatcher.
+    fn read_ranges(
+        &self,
+        ranges: Vec<(MessageId, ByteRange)>,
+    ) -> BoxFuture<'static, VortexResult<Vec<(MessageId, Bytes)>>> {
+        let reader = self.input.clone();
+
+        let result_rx = self
+            .dispatcher
+            .dispatch(move || async move { read_ranges(reader, ranges).await })
+            .vortex_expect("dispatch async task");
+
+        result_rx
+            .map(|res| match res {
+                Ok(result) => result,
+                Err(_) => vortex_bail!("dispatcher channel canceled"),
```

Comment:
sure, though `Canceled` is the only error variant for this one.

I added it to the vortex_bail statement, since `with_context` is only defined for VortexErrors